Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayer.cpp

```diff
@@ -539,10 +539,8 @@ void CoordinatedPlatformLayer::setDirtyRegion(Vector<IntRect, 1>&& dirtyRegion)
 void CoordinatedPlatformLayer::setDamage(Damage&& damage)
 {
     ASSERT(m_lock.isHeld());
-    if (m_damage == damage)
-        return;
-
-    m_damage = WTFMove(damage);
+    if (m_damage != damage)
+        m_damage = WTFMove(damage);
     m_pendingChanges.add(Change::Damage);
```

Comment:
You always want to add `Change::Damage` to `m_pendingChanges`, even if `damage` wasn't changed?

Reply:
Correct. The current implementation of damage handling in `TextureMapperLayer` relies on the fact that damage is set every frame, even if it haven't changed between frames. And this code here is being triggered by `GraphicsLayerCoordinated::updateDirtyRegion()` that is called only if `DirtyRegion` changed. So atm. that's expected.